### ISSUE TITLE
Add custom sunrise and sunset times

### ISSUE THREAD

**Issue by allanlaal** [jonls/redshift#9]
It would be extremely useful to set a parameter with custom sunrise and sunset times instead of a location.
I live in Estonia, where the sun sets in winter at around 15:00 and in summer around 22:00. I dont want my laptops screen temperature to tell my brain to start preparing for sleep at 15:00. This can be solved with a custom forced sunset time. This new parameter would override the one set with "-l"

**Comment by viljoviitanen** [comment]
Just in case someone is interested in running a "fixed" time redshift, and haven't figured out an easy way to do it on an old version of redshift (a lot of people, myself included are running 1.8 because it's in ubuntu 14.04!) set it to some location at the equator, there the sunset is always about the same time.

E.g. in Finland the following configuration results a "sunset" at 19.48-20.18 local time all year round (GMT+2/standard time, during daylight saving it's one hour more):

    $ cat .config/redshift.conf 
    [redshift]
    location-provider=manual
    [manual]
    lat=0.0
    lon=0.0

**Comment by linusrachlis** [comment]
I had a similar desire and realized I could roll my own solution using crontab:

```
# redshift - full spectrum at 7am and redshift at 10pm
0	7	*	*	*	env DISPLAY=:0 redshift -O 6500
0	22	*	*	*	env DISPLAY=:0 redshift -O 5000
```

Had to add the DISPLAY env to make it work, which I picked up from https://help.ubuntu.com/community/CronHowto.

Of course, this causes a jarring transition, but I could make it change gradually by adding more entries.

**Comment by linusrachlis** [comment]
Finally got this cron script working the way I want. If interested, see https://github.com/linusrachlis/redshift-continuous

It runs every 5 minutes, and uses Redshift to update the screen temperature based on the time. I've set it up to shift over a 1-hour period - dawn from 7-8am, and dusk from 9-10pm. It's easy to change the script though.

Not sure how portable it will be -- for all I know, the DISPLAY env var may be Ubuntu only, I don't know. (See previous post.)

The good thing is, it doesn't matter much if my computer was off for the transition time, because in <5 minutes after I wake it up, it will adjust the temp to match the time of day.

HTH!

**Comment by lacombar** [comment]
This is full of horrible hack... This is the kind of stupid feature that make me want to use an OS X desktop to be able to use f.lux...

**Comment by motersen** [comment]
@lacombar opinions about software architecture aside, why don't you use f.lux for Linux?

**Comment by draeath** [comment]
Checking in to say there's still users interested in at least offsetting the time calculation without having to do it ourselves (as in these cron examples) or abuse the elevation-low or elevation-high parameters. A simple "time-offset" parameter that pushes the curve one way or the other would be helpful! It would be trivial for an end-user to calculate this (eg just find local sunrise or sunset time, and compute the difference between it and their typical wake/rest times).

I might take up writing a patch myself, though c/c++ are not something I'm practiced with, if this would be helpful.

**Comment by jonls** [comment]
Implemented in #529

**Comment by louisgag** [comment]
I'm not sure if this is the proper thread for this, but I also wished to manually control screen brightness and temperature using keyboard shortcuts and couldn't find any simple way to do it so I made a small wrapper bash script to [manually control redshift](https://github.com/louisgag/redshiftAdjust).
